### PR TITLE
Add support for multiple nem files

### PR DESCRIPTION
**Pull request by Kyle-Verhoog**
The closer the nem file is to the current directory the
higher precedence it has for finding, creating and deleting
entries.

### CODE REVIEW

File: nem.rs [hunk 1]
```diff
@@ -76,10 +133,32 @@ fn mnemonic(v: &[String], existing: Vec<&String>) -> String {
...
-    nem_file.file_name = String::from(".nem.toml");
-    nem_file.sort();
+
+    let mut nem_files = NemFiles {
+        nem_files: Vec::new(),
+    };
+    let mut path_buf = env::current_dir().expect("Couldn't get CWD");
+    loop {
+        path_buf.push(".nem.toml");
+        let path = path_buf.as_path();
```
Comment (p7g) [inline nem.rs:143]: I think you don't have to make a `Path` object from the `PathBuf`, most of the methods seem to accept `AsRef<Path>`, which a `PathBuf` implements.

File: nem.rs [hunk 2]
```diff
@@ -76,10 +133,32 @@ fn mnemonic(v: &[String], existing: Vec<&String>) -> String {
...
+        if path.exists() {
+            match fs::read_to_string(&path) {
+                Ok(contents) => {
+                    let mut nem_file: NemFile =
+                        toml::from_str(&contents).expect("Couldn't parse toml file");
+                    nem_file.file_name = String::from(path.to_str().unwrap());
+                    nem_file.sort();
+                    nem_files.nem_files.push(nem_file);
+                }
+                Err(_) => {}
```
Comment (p7g) [inline nem.rs:153]: Maybe would be worth `eprintln!`-ing the error? Every `Error` in rust has to also implement `Display`, so you can format it with `"{}"`

**Review by p7g — APPROVED** [review]
:ok_hand: 

What if you're in a directory that isn't a child of the home directory? Should there be special cases like `~/.nem.toml` or `/etc/nem/nem.toml`?

File: nem.rs [hunk 3]
```diff
@@ -62,6 +62,63 @@ struct Entry {
...
+            .iter_mut()
+            .find(|nf| nf.find_entry_by_code(code).is_some())
+            .and_then(|nf| nf.rm_entry(code))
+    }
+
+    fn find_entry_by_code(&self, code: &String) -> Option<&Entry> {
+        self.nem_files
+            .iter()
+            .find(|nf| nf.find_entry_by_code(code).is_some())
+            .and_then(|nf| nf.find_entry_by_code(code))
```
[the remark this answers is not in the document]
Comment (Kyle-Verhoog) [reply nem.rs:105]: yesss i was looking for a better way 🙌

**Comment by Kyle-Verhoog** [comment]
Yes I believe the python implementation explicitly checked the home directory and ~/.config/.nem.toml. We should do a follow up with support for those and whatever other common config paths might exist 👍

**Pull request merged**